Add Reindex command to reflect direct db changes into Elastic index (#311)
humanmade/altis-dev-tools: docs/testing-with-codeception.md

```sh
composer dev-tools codecept run -- -c path/to/codeception.yml
```

## FAQ

- Why do my tests fail because it cannot find the content I created using WPBrowser DB helper functions ?

> WPBrowser DB helper functions, like `$I->havePostInDatabase()` and the alike, uses direct database queries to manage content, which means WordPress filters are not run for those operations, so integrations like ElasticPress are not notified of the changes and do not update the Elastic index as a result of that. So while the content is created in the database, it is not synced to Elastic index, and subsequently will not show up in queries that are typically run by Elastic.

Reviewer: ```suggestion
WPBrowser DB helper functions, like `$I->havePostInDatabase()` and the like, use direct database queries to manage content, which means WordPress filters are not run for those operations. This means that integrations like ElasticPress are not notified of the changes and do not update the Elasticsearch index as a result of that. So while the content is created in the database, it is not synced to Elasticsearch, and subsequently will not show up in queries that are handled by ElasticPress.
```

Nitpick but "Elastic" can be a bit broad in infrastructure terms as there are things like Elasticache (Redis) that may get discussed too. Better to be explicit I think.

Author: I had been under the impression the product changed name to Elastic rathe than Elasticsearch, going through docs now that seems to be wrong 🤦 . Rest of changes are legit 👍 got it.

Reviewer: Ohh sorry yeah - the company behind it did though the software is still Elasticsearch, and to confuse matters further AWS's fork is called OpenSearch!

Author: Yeah, that's what confused me then!

> The fix for that would be to explicitly reindex content after such direct database operations to ensure the Elastic index is synced properly, and for that we have `$I->reindexContent()` helper function that does that for you.

Reviewer: ```suggestion
The fix for this is to explicitly reindex content after such direct database operations to ensure the Elasticsearch index is synced properly, and for that you can use the `$I->reindexContent()` helper function.
```

Example:

```php
$I->havePostInDatabase( $params );
$I->haveUserInDatabase( $params );

> $I->reindexContent();

Reviewer: Can we document the parameter it takes so folks know you could pass additional arguments like `--indexables` etc?

Author: ```suggestion
// Use $extra_params to pass params to the `elasticpress` CLI command like `--indexables=post,user`, etc.
$I->reindexContent( $extra_params );
```
